src/Dotnet.Script.Tests/ScriptExecutionTests.cs

```diff
@@ -192,6 +200,25 @@ public void ShouldSupportInlineNugetReferencesWithTrailingSemicoloninEvaluatedCo
             Assert.Contains("AutoMapper.MapperConfiguration", result.output);
         }
 
+        [Fact]
+        public static void ShouldHandleIssue235()
```

Comment:
I am not sure I understand what is this trying to test?
so it compiles a script into a DLL, and then runs a script that references that DLL - what conditions does that create? that certain DLL dependency is loaded twice?

Reply:
1. In a script I call Mapper.CreateNew(), so that NuGet Assembly would be loaded,
2. I reference dependency dll, that also references that Nuget just with a different version
3. Execute referenced dlls method and dotnet-script tries to redirect outdated nuget assembly to a new one.
And the test is red before my changes.

I compile script to dll, because it was easiest solution for me to reproduce this behavior.

Reply:
I know this isn't a clean solution :). I was hoping that if I do this you will point me in a better direction ;).

Reply:
OK I see what you mean. Yeah there could be a simpler way to do it, but it's OK for me for now

Reply:
@SlowLogicBoy , @filipw  Maybe a simpler approach would be to use the `dotnet cli` to create your test dll?

Something like 
```
using (var disposableFolder = new DisposableFolder())
{
    var projectFolder = Path.Combine(disposableFolder.Path, "TestLibrary");
    ProcessHelper.RunAndCaptureOutput("dotnet", new[] {"new classlib -n TestLibrary"}, disposableFolder.Path);
    ProcessHelper.RunAndCaptureOutput("dotnet", new[] { "add TestLibrary.csproj package AgileObjects.AgileMapper -v 0.23.0" }, projectFolder);
    ProcessHelper.RunAndCaptureOutput("dotnet", new[] {"build -c release -o ./"}, projectFolder);
}
```

Reply:
I was thinking about making and pushing .csproj itself, but I couldn't find appropirate directory for it, maybe this disposable folder is a good idea, but wouldn't it be cheaper to just create csproj and push it, instead of creating a new one everytime?

Reply:
Performance in the test project is not all that important so you should be good creating it every time 👍

Reply:
@SlowLogicBoy Take a look at this. I think this pretty much tests the same as the original code.

```
public static void ShouldHandleIssue235()
{
    string code = 
    @"using AgileObjects.AgileMapper;
    public class TestClass
    {
        public TestClass()
        {
            IMapper mapper = Mapper.CreateNew();
        }
    }";

    string script =
    @"#! ""netcoreapp2.0""
    #r ""nuget: AgileObjects.AgileMapper, 0.23.1""
    #r ""TestLibrary.dll""
    
    using AgileObjects.AgileMapper;

    IMapper mapper = Mapper.CreateNew();
    var testClass = new TestClass();
    Console.WriteLine(""Hello World!"");";


    using (var disposableFolder = new DisposableFolder())
    {
        var projectFolder = Path.Combine(disposableFolder.Path, "TestLibrary");
        ProcessHelper.RunAndCaptureOutput("dotnet", new[] {"new classlib -n TestLibrary"}, disposableFolder.Path);
        ProcessHelper.RunAndCaptureOutput("dotnet", new[] { "add TestLibrary.csproj package AgileObjects.AgileMapper -v 0.23.0" }, projectFolder);
        File.WriteAllText(Path.Combine(projectFolder, "Class1.cs"), code);
        File.WriteAllText(Path.Combine(projectFolder, "script.csx"), script);
        ProcessHelper.RunAndCaptureOutput("dotnet", new[] {"build -c release -o ./"}, projectFolder);

        var dotnetScriptArguments = GetDotnetScriptArguments(Path.Combine(projectFolder, "script.csx"));
        var result = ProcessHelper.RunAndCaptureOutput("dotnet", dotnetScriptArguments);
        Assert.Contains("Hello World!", result.output);
    }
}
```

Reply:
👍 This works and way cleaner thanks!